e2e/deployers/retry.go

```diff
@@ -68,3 +76,88 @@ func WaitWorkloadHealth(ctx types.TestContext, cluster types.Cluster, namespace
 		}
 	}
 }
+
+func WaitForResourceDelete(
+	ctx types.Context,
+	cluster types.Cluster,
+	key k8stypes.NamespacedName,
+	obj client.Object,
+) error {
+	log := ctx.Logger()
+	resourceType := getResourceTypeName(obj)
+	startTime := time.Now()
+
+	log.Debugf("Waiting until %s \"%s/%s\" is deleted in cluster %q",
+		resourceType, key.Namespace, key.Name, cluster.Name)
+
+	for {
+		if err := cluster.Client.Get(ctx.Context(), key, obj); err != nil {
+			if !k8serrors.IsNotFound(err) {
+				return err
+			}
+
+			log.Debugf("%s \"%s/%s\" deleted in cluster %q",
+				resourceType, key.Namespace, key.Name, cluster.Name)
+
+			return nil
+		}
+
+		if time.Since(startTime) > 60*time.Second {
+			return fmt.Errorf("timeout deleting %s \"%s/%s\" in cluster %q",
+				resourceType, key.Namespace, key.Name, cluster.Name)
+		}
+
+		if err := util.Sleep(ctx.Context(), time.Second); err != nil {
+			return err
+		}
+	}
+}
+
+func WaitApplicationSetDeleted(ctx types.TestContext) error {
```

Comment:
We want to have consistent signature for these helpers:

    WaitForXXXDelete(ctx types.Context, cluster types.Cluster, namespacel, name string, deadline time.Time)